lms/static/scripts/frontend_apps/components/URLPicker.js

```diff
@@ -20,23 +21,52 @@ import Dialog from './Dialog';
 export default function URLPicker({ onCancel, onSelectURL }) {
   const input = useRef(/** @type {HTMLInputElement|null} */ (null));
   const form = useRef(/** @type {HTMLFormElement|null} */ (null));
+  // Is there a validation error message to show?
+  const [showValidationError, setValidationError] = useState(false);
+  // The actual validation error message.
+  const [validationMessage, setValidationMessage] = useState('');
 
   /** @param {Event} event */
   const submit = event => {
     event.preventDefault();
+
     if (form.current.checkValidity()) {
       onSelectURL(input.current.value);
     } else {
-      form.current.reportValidity();
+      setValidationMessage('A valid URL is required');
+      setValidationError(true);
     }
   };
 
+  /**
+   * If any input is detected, close the ValidationMessage.
+   */
+  const handleKeyDown = () => {
+    setValidationError(false);
+  };
+
+  const submitButton = (
+    <span className="URLPicker__buttons">
+      {validationMessage && (
```

Comment:
This code is mostly just borrowed from SubmitGradeForm, we have to do a tiny bit of trickery here and check if the message exists the very first time things are rendered to prevent the animation from quickly showing, and then collapsing the red valitation banner. For this reason it helps to keep tract of both the open/closed state and a message state which is initially a blank string.